stripe/src/main/java/com/stripe/android/model/CardBrand.kt

```diff
@@ -74,7 +79,13 @@ enum class CardBrand(
         "jcb",
         "JCB",
         R.drawable.stripe_ic_jcb,
-        pattern = Pattern.compile("^(35)[0-9]*$")
+        pattern = Pattern.compile("^(35)[0-9]*$"),
+        partialPatterns = mapOf(
+            2 to Pattern.compile("^(35)$"),
+            3 to Pattern.compile("^(35[2-8])$"),
+            4 to Pattern.compile("^(352[89]|35[3-8][0-9])$"),
+            5 to Pattern.compile("^(3528[0-9]|3529[0-9]|35[3-8][0-9]{2})$")
```

Comment:
We can actually simplify regexes 3-5 to `^(352[89]|35[3-8])`, since it's definitely a JCB card after we see the first 3/4 characters.

The JCB range is 352800 to 358999. If you expand them, we only need these prefixes:
```
3528
3529
353
354
355
356
357
358
```